### ISSUE TITLE
3.2.8 breaks my extend process

### ISSUE THREAD

**Issue by kud** [sass/sass#722]
Hey,

I don't know what you changed but my extend doesn't work correctly anymore.

```
#menu {
    li {
        a {
            display: block;
            line-height: 38px;
            cursor: pointer;
            font-size: 14px;
            font-weight: bold;
            text-shadow: 0 -1px black;
            color: #bbb;
            text-transform: uppercase;
            padding: 2px 11px 0 12px;

            &:hover,
            &:active {
                text-shadow: 0 1px #fff;
                color: #000;
                border-bottom: 1px solid #a6a6a6;
                background: #FBFBFB;
                @include background(linear-gradient(#FBFBFB, #E4E4E4));
            }

            &:active {
                background: #E4E4E4;
                @include background(linear-gradient(#E4E4E4, #FBFBFB));
            }

            &.is-active {
                font-size: 14px;
                font-weight: bold;
                text-shadow: 0 -1px black;
                color: #bbb;
                text-transform: uppercase;
                background: none;
                color: #fff;
                border: 0;

                &:hover,
                &:active {
                    font-size: 14px;
                    font-weight: bold;
                    text-shadow: 0 -1px black;
                    color: #bbb;
                    text-transform: uppercase;
                    background: none;
                    color: #fff;
                    border: 0;
                }

                &:before {
                    @include opacity(1);
                }
            }

            &:before {
                content: "";
                @include inline-block();
                zoom: 1;
                margin-right: 10px;
                vertical-align: top;
                @include opacity(0.7);
            }

            &:hover:before {
                @include opacity(1);
            }
        }

        &.home {
            a {
                &:before {
                    @extend .icons-menu-home;
                    margin-top: 9px;
                    margin-right: 0px;
                }

                &:hover:before {
                    @extend .icons-menu-homehover;
                }

                &.is-active:hover:before {
                  font-size: 12;
                  @extend .icons-menu-home;
                }
            }
        }

        &.tv {
            a {
                &:before {
                    @extend .icons-menu-tv;
                    margin-top: 11px;
                }

                &:hover:before {
                    @extend .icons-menu-tvhover;
                }

                &.is-active:hover:before {
                    @extend .icons-menu-tv;
                }
            }
        }

        &.progtv {
            a {
                &:before {
                    @extend .icons-menu-progtv;
                    margin-top: 9px;
                }

                &:hover:before {
                    @extend .icons-menu-progtvhover;
                }

                &.is-active:hover:before {
                    @extend .icons-menu-progtv;
                }
            }
        }

        &.replay {
            a {
                &:before {
                    @extend .icons-menu-replay;
                    margin-top: 12px;
                }

                &:hover:before {
                    @extend .icons-menu-replayhover;
                }

                &.is-active:hover:before {
                    @extend .icons-menu-replay;
                }
            }
        }

        &.blog {
            a {
                &:before {
                    @extend .icons-menu-blog;
                    margin-top: 10px;
                }

                &:hover:before {
                    @extend .icons-menu-bloghover;
                }

                &.is-active:hover:before {
                    @extend .icons-menu-blog;
                }
            }
        }

        &.livetweet {
            a {
                &:before {
                    @extend .icons-menu-livetweet;
                    margin-top: 12px;
                }

                &:hover:before {
                    @extend .icons-menu-livetweethover;
                }

                &.is-active:hover:before {
                    @extend .icons-menu-livetweet;
                }
            }
        }
    }
}
```

Everything works fine except these kind of lines:

```
&.is-active:hover:before {
    @extend .icons-menu-livetweet;
}
```

This class comes from the magic code for generating sprite.

With 3.2.7, it works like a charm. With 3.2.8, it doesn't render the .is-active:hover:before.

**Comment by MoOx** [comment]
Live example here http://sassmeister.com/gist/4760632
If you comment the &.before extend, it sort of works.

**Comment by Rydgel** [comment]
May be a regression introduced with the fix of #511 https://github.com/nex3/sass/commit/3700884c8c150e61da8d24168d25db34fccbd19b

I reproduce this bug as well.
